cpp/include/cudf/detail/gather.cuh

```diff
@@ -522,14 +521,18 @@ void gather_bitmask(table_view const& source,
   }();
   impl(*device_source,
        gather_map,
-       masks,
+       d_target_masks.data(),
        target.size(),
        target_rows,
-       d_valid_counts.data().get(),
+       d_valid_counts.data(),
        stream);
 
   // Copy the valid counts into each column
-  auto const valid_counts = thrust::host_vector<size_type>(d_valid_counts);
+  std::vector<size_type> valid_counts(d_valid_counts.size());
+  CUDA_TRY(cudaMemcpy(valid_counts.data(),
```

Comment:
Right. This struck me as odd as well. `cudaMemcpyAsync` + `stream.synchronize()`.